xds/src/test/java/io/grpc/xds/XdsNameResolverTest.java

```diff
@@ -64,84 +92,82 @@ public void uncaughtException(Thread t, Throwable e) {
           .setServiceConfigParser(mock(ServiceConfigParser.class))
+          .setScheduledExecutorService(fakeClock.getScheduledExecutorService())
           .setChannelLogger(mock(ChannelLogger.class))
           .build();
 
-  private final XdsNameResolverProvider provider = new XdsNameResolverProvider();
 
-  @Mock private NameResolver.Listener2 mockListener;
+  private final Queue<StreamObserver<DiscoveryResponse>> responseObservers = new ArrayDeque<>();
 
-  @Test
-  public void validName_withAuthority() {
-    XdsNameResolver resolver =
-        provider.newNameResolver(
-            URI.create("xds-experimental://trafficdirector.google.com/foo.googleapis.com"), args);
-    assertThat(resolver).isNotNull();
-    assertThat(resolver.getServiceAuthority()).isEqualTo("foo.googleapis.com");
-  }
+  @Mock
+  private BackoffPolicy.Provider backoffPolicyProvider;
+  @Mock
+  private NameResolver.Listener2 mockListener;
 
-  @Test
-  public void validName_noAuthority() {
-    XdsNameResolver resolver =
-        provider.newNameResolver(URI.create("xds-experimental:///foo.googleapis.com"), args);
-    assertThat(resolver).isNotNull();
-    assertThat(resolver.getServiceAuthority()).isEqualTo("foo.googleapis.com");
-  }
+  private XdsChannelFactory channelFactory;
+  private XdsNameResolver xdsNameResolver;
 
-  @Test
-  public void invalidName_hostnameContainsUnderscore() {
-    try {
-      provider.newNameResolver(URI.create("xds-experimental:///foo_bar.googleapis.com"), args);
-      fail("Expected IllegalArgumentException");
-    } catch (IllegalArgumentException e) {
-      // Expected
-    }
-  }
+  @Before
+  public void setUp() throws IOException {
+    final String serverName = InProcessServerBuilder.generateName();
+    AggregatedDiscoveryServiceImplBase serviceImpl = new AggregatedDiscoveryServiceImplBase() {
+      @Override
+      public StreamObserver<DiscoveryRequest> streamAggregatedResources(
+          final StreamObserver<DiscoveryResponse> responseObserver) {
+        responseObservers.offer(responseObserver);
+        @SuppressWarnings("unchecked")
+        StreamObserver<DiscoveryRequest> requestObserver = mock(StreamObserver.class);
+        return requestObserver;
+      }
+    };
 
-  @Test
-  public void resolve_bootstrapResult() {
-    final ChannelCreds loasCreds = new ChannelCreds("loas2", null);
-    final ChannelCreds googleDefaultCreds = new ChannelCreds("google_default", null);
+    cleanupRule.register(
+        InProcessServerBuilder
+            .forName(serverName)
+            .addService(serviceImpl)
+            .directExecutor()
+            .build()
+            .start());
+    final ManagedChannel channel =
+        cleanupRule.register(InProcessChannelBuilder.forName(serverName).directExecutor().build());
+
+    channelFactory = new XdsChannelFactory() {
+      @Override
+      ManagedChannel createChannel(List<ServerInfo> servers) {
+        return channel;
```

Comment:
```java
assertThat(Iterables.getOnlyElement(servers).getServerUri())
            .isEqualTo(serverName);
```

or create the inprocess `channel` here with the server name from `servers`

Reply:
Done.